src/services/codefixes/importFixes.ts

```diff
@@ -774,7 +775,7 @@ namespace ts.codefix {
 
         const isValidTypeOnlyUseSite = isValidTypeOnlyAliasUseSite(symbolToken);
         const useRequire = shouldUseRequire(sourceFile, program);
-        const exportInfos = getExportInfos(symbolName, getMeaningFromLocation(symbolToken), cancellationToken, sourceFile, program, useAutoImportProvider, host, preferences);
+        const exportInfos = getExportInfos(symbolName, isJSXTagName(symbolToken), getMeaningFromLocation(symbolToken), cancellationToken, sourceFile, program, useAutoImportProvider, host, preferences);

```

Comment:
I don't understand why jsx tag names are unconditionally capitalised. Is it because the convention is for a CamelCase component to be default exported by a snake-case module, and it's assuming the convention holds?

Edit: Or is it because of the filter code you introduced in collectAutoImports?

Reply:
It’s because JSX elements that begin with a lowercase letter _must_ be intrinsics (e.g. HTML tag names). However, many people kebab-case their filenames, which means we’ll produce a lowercase identifier for the default export if we can’t find another suitable name.

Reply:
By way of example,

```ts
import myComponent from "./my-component";

(<myComponent />);
```

does not type check, even if `myComponent` resolves to a valid component declaration.